Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -782,29 +803,24 @@ public void SessionIntegrationTest_ProgressObservable(ProgressMode mode)
                     {
                         callbacksInvoked++;
 
-                        if (p.TransferredBytes > p.TransferableBytes)
+                        if (p.ProgressEstimate < 0.0 || p.ProgressEstimate > 1.0)
                         {
-                            // TODO https://github.com/realm/realm-dotnet/issues/2360: this seems to be a regression in Sync.
-                            // throw new Exception($"Expected: {p.TransferredBytes} <= {p.TransferableBytes}");
-                        }
-
-                        if (mode == ProgressMode.ForCurrentlyOutstandingWork)
-                        {
-                            if (p.TransferableBytes <= objectSize ||
-                                p.TransferableBytes >= (objectsToRecord + 2) * objectSize)
-                            {
-                                throw new Exception($"Expected: {p.TransferableBytes} to be in the ({objectSize}, {(objectsToRecord + 1) * objectSize}) range.");
-                            }
+                            throw new Exception($"Expected progress estimate to be between 0.0 and 1.0, but was {p.ProgressEstimate}");
                         }
                     }
                     catch (Exception e)
                     {
                         completionTcs.TrySetException(e);
                     }
 
-                    if (p.TransferredBytes >= p.TransferableBytes)
+                    if (p.IsComplete)
                     {
-                        completionTcs.TrySetResult(p.TransferredBytes);
+                        if (p.ProgressEstimate != 1.0)
+                        {
+                            throw new Exception($"Expected progress estimate to be complete if and only if ProgressEstimate == 1.0");
```

Comment:
This is outside the try-catch, so if we get here, we won't set the completionTcs exception and instead will throw it into the void.